algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableTrialFactory.java

```diff
@@ -50,21 +60,31 @@ public Set<ActionableEntry> create(@NotNull CkbEntry entry, @NotNull String sour
-
-                    if (cancerTypeExtraction != null) {
-                        actionableTrials.add(ImmutableActionableEntry.builder()
-                                .source(Knowledgebase.CKB_TRIAL)
-                                .sourceEvent(sourceEvent)
-                                .sourceUrls(Sets.newHashSet("https://clinicaltrials.gov/study/" + trial.nctId()))
-                                .treatment(ImmutableTreatment.builder().name(trial.title()).build())
-                                .applicableCancerType(cancerTypeExtraction.applicableCancerType())
-                                .blacklistCancerTypes(cancerTypeExtraction.blacklistedCancerTypes())
-                                .level(EvidenceLevel.B)
-                                .direction(EvidenceDirection.RESPONSIVE)
-                                .evidenceUrls(countries)
-                                .build());
+                for (Therapy therapy : trial.therapies()) {
+                    String therapyName = therapy.therapyName();
+                    for (Indication indication : trial.indications()) {
+                        CancerTypeExtraction cancerTypeExtraction = ActionableFunctions.extractCancerTypeDetails(indication);
+                        if (cancerTypeExtraction != null) {
+                            if (!blacklistStudy.isBlacklistStudy(trial.nctId(), therapyName, cancerTypeExtraction.applicableCancerType().name(),
+                                    sourceGene, sourceEvent)) {
+                                actionableTrials.add(ImmutableActionableEntry.builder()
+                                        .source(Knowledgebase.CKB_TRIAL)
+                                        .sourceEvent(sourceEvent)
+                                        .sourceUrls(Sets.newHashSet("https://ckbhome.jax.org/clinicalTrial/show?nctId=" + trial.nctId()))
+                                        .intervention(ImmutableClinicalTrial.builder()
+                                                        .studyNctId(trial.nctId())
+                                                        .studyTitle(trial.title())
+                                                        .countriesOfStudy(countries)
+                                                        .therapyName(therapyName)
+                                                        .build())
+                                        .applicableCancerType(cancerTypeExtraction.applicableCancerType())
+                                        .blacklistCancerTypes(cancerTypeExtraction.blacklistedCancerTypes())
+                                        .level(EvidenceLevel.B)
+                                        .direction(EvidenceDirection.RESPONSIVE)
+                                        .evidenceUrls(Sets.newHashSet("https://clinicaltrials.gov/study/" + trial.nctId()))
+                                        .build());
+                            }
+                        }
```

Comment:
This logic will lead to a lot of additional trials potentially. While previous there would be 1 trial for N therapies, there will now be N trials. 

Did you consider adding a Set<String> therapyNames to a single ClinicalTrial instead?

This change has quite big implications at least, would probably need decent adjustment in ACTIN for example, if we want to collapse trials again.

Reply:
I hadn't consider that, but I have update it to 1 trial for N therapies. The N therapies has effect on the different arms of the studies. The combined therapies for a study is defined by therapyA + therapyB in a single string